base/pkg/libgit2/strarray.jl

```diff
@@ -0,0 +1,25 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+function StrArrayStruct{T<:AbstractString}(strs::T...)
+    count = length(strs)
+    strings = convert(Ptr{Ptr{Uint8}}, Libc.malloc(sizeof(Ptr{Uint8}) * count))
+    for i=1:count
+        len = length(strs[i])
+        #in_ptr  = convert(Ptr{Uint8}, bytestring(strs[i]))
+        in_ptr  = pointer(bytestring(strs[i]))
+        out_ptr = convert(Ptr{Uint8}, Libc.malloc(sizeof(Uint8) * (len + 1)))
+        unsafe_copy!(out_ptr, in_ptr, len)
+        unsafe_store!(out_ptr, zero(Uint8), len + 1) # NULL byte
+        unsafe_store!(strings, out_ptr, i)
+    end
+    return StrArrayStruct(strings, count)
+end
+StrArrayStruct{T<:AbstractString}(strs::Vector{T}) = StrArrayStruct(strs...)
```

Comment:
Seems like this should be the other way around: your "main" constructor should be the `Vector{T}